velox/exec/Unnest.cpp

```diff
@@ -191,7 +248,40 @@ const Unnest::UnnestChannelEncoding Unnest::generateEncodingForChannel(
   // Make dictionary index for elements column since they may be out of order.
   vector_size_t index = 0;
   bool identityMapping = true;
-  for (auto row = start; row < start + size; ++row) {
+
+  const auto firstLastRowGenerator =
+      [&](vector_size_t row, vector_size_t start, vector_size_t end) {
+        if (!currentDecoded.isNullAt(row)) {
```

Comment:
Is this a copy-paste from the logic for middle rows? What are the differences? Is there a way to avoid copy-paste?

Reply:
No, it not just a copy paste, the identityMapping condition, unnest size condition, and number of null value are different, if we unify them, extra condition check will be introduced to middle rows.

Reply:
For the readability, I will consolidate them.